vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -49,30 +43,30 @@ def format_metric_name(counter):
     )
 
 
-def match_any_regex(string, regexes):
-    # type: (str, List[Pattern]) -> bool
-    for regex in regexes:
-        match = regex.match(string)
-        if match:
-            return True
-    return False
-
-
-def is_resource_collected_by_filters(mor, infrastructure_data, resource_filters, tags_cache):
-    # type: (vim.ManagedEntity, InfrastructureData, ResourceFilters, TagsCache) -> bool
+def is_resource_collected_by_filters(mor, infrastructure_data, resource_filters, resource_tags):
+    # type: (vim.ManagedEntity, InfrastructureData, List[ResourceFilter], List[str]) -> bool
     resource_type = MOR_TYPE_AS_STRING[type(mor)]
-
-    if not [f for f in resource_filters if f.resource == resource_type and f.is_whitelist]:
+    # Limit filters to those for the resource_type of the mor
+    resource_filters = [f for f in resource_filters if f.resource_type == resource_type]
+
+    whitelist_filters = [f for f in resource_filters if f.is_whitelist]
+    blacklist_filters = [f for f in resource_filters if not f.is_whitelist]
+
+    match_whitelist = False
+    match_blacklist = False
+    for resource_filter in whitelist_filters:
+        if resource_filter.match(mor, infrastructure_data, resource_tags):
+            match_whitelist = True
+            break
+    for resource_filter in blacklist_filters:
+        if resource_filter.match(mor, infrastructure_data, resource_tags):
+            match_blacklist = True
+            break
```

Comment:
Shall we match first by `blacklist first` (bfore whitelist) and exit right away if there is a match?

Reply:
Yeah good idea 👍